common/src/test/java/io/netty/util/RecyclerTest.java

```diff
@@ -24,7 +24,7 @@
 
 public class RecyclerTest {
 
-    @Test(expected = IllegalStateException.class)
+    //@Test(expected = IllegalStateException.class)
```

Comment:
@ninja- pretty sure @normanmaurer would rather you leave all the safety checks in place (even without a property to disable them) unless you can provide benchmark results showing that with the safety checks disabled you see a notable performance gain..

Perhaps benchmark results between old recycler and the "simplified recycler" and then finally benchmark results between the "simplified recycler" and the "simplified recycler" with safety checks disabled.